internal/eval/evalers.go

```diff
@@ -1218,7 +1305,299 @@ func newExtensionEval(name types.Path, args []Evaler) Evaler {
 			return newIPTestEval(args[0], ipTestMulticast)
 		case name == "isInRange":
 			return newIPIsInRangeEval(args[0], args[1])
+
+		case name == "toDate":
+			return newToDateEval(args[0])
+		case name == "toTime":
+			return newToTimeEval(args[0])
+		case name == "toMilliseconds":
+			return newToMillisecondsEval(args[0])
+		case name == "toSeconds":
+			return newToSecondsEval(args[0])
+		case name == "toMinutes":
+			return newToMinutesEval(args[0])
+		case name == "toHours":
+			return newToHoursEval(args[0])
+		case name == "toDays":
+			return newToDaysEval(args[0])
+
+		case name == "offset":
+			return newOffsetEval(args[0], args[1])
+		case name == "durationSince":
+			return newDurationSinceEval(args[0], args[1])
 		}
 	}
 	return newErrorEval(fmt.Errorf("%w: %s", errUnknownExtensionFunction, name))
 }
+
+// comparableValueLessThanEval struct
+type comparableValueLessThanEval struct {
+	lhs Evaler
+	rhs Evaler
+}
+
+func newComparableValueLessThanEval(lhs Evaler, rhs Evaler) *comparableValueLessThanEval {
+	return &comparableValueLessThanEval{
+		lhs: lhs,
+		rhs: rhs,
+	}
+}
+
+func (n *comparableValueLessThanEval) Eval(env *Env) (types.Value, error) {
+	lhs, err := evalComparableValue(n.lhs, env)
+	if err != nil {
+		return zeroValue(), err
+	}
+	rhs, err := evalComparableValue(n.rhs, env)
+	if err != nil {
+		return zeroValue(), err
+	}
+	ok, err := lhs.LessThan(rhs)
+	if err != nil {
+		return types.False, fmt.Errorf("%w: %w", ErrType, err)
+	}
+	return types.Boolean(ok), nil
+}
+
+var _ Evaler = &comparableValueLessThanEval{}
+
+// comparableValueGreaterThanEval struct
+type comparableValueGreaterThanEval struct {
+	lhs Evaler
+	rhs Evaler
+}
+
+func newComparableValueGreaterThanEval(lhs Evaler, rhs Evaler) *comparableValueGreaterThanEval {
+	return &comparableValueGreaterThanEval{
+		lhs: lhs,
+		rhs: rhs,
+	}
+}
+
+func (n *comparableValueGreaterThanEval) Eval(env *Env) (types.Value, error) {
+	lhs, err := evalComparableValue(n.lhs, env)
+	if err != nil {
+		return zeroValue(), err
+	}
+	rhs, err := evalComparableValue(n.rhs, env)
+	if err != nil {
+		return zeroValue(), err
+	}
+	ok, err := lhs.LessThanOrEqual(rhs)
```

Comment:
tiny nit: the `ok` here maybe should be `isLessThanOrEqual`? `ok` implies something else to me.